web/src/stream_list.ts

```diff
@@ -702,6 +702,9 @@ export function update_dom_unread_counts_visibility(): void {
         const stream_counts = unread.unread_count_info_for_stream(stream.sub.stream_id);
         toggle_hide_unread_counts($subscription_block, is_muted, stream_counts.unmuted_count);
     }
+    // Updates the count visibilties for folder headers and inactive toggles.
+    // TODO: that logic should ideally be moved to this file.
```

Comment:
I don't think we need this TODO; it seems to me this function should just be calling `build_stream_list` with `force_rerender`, and we don't need all this custom code that fetches all the unread count data multiple times?